From 2a24482c1a3fd1e344312bec4f02c41def7a5b05 Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Fri, 6 Mar 2020 14:21:52 +0800
Subject: [PATCH] phosphor-dump-manager: fix SEGV exception

In the iteration of deleting std::map entry, should not
execute ++ operation on the entry that has has been deleted.

Error:
xyz.openbmc_project.Dump.Manager.service: Main process exited, code=dumped, status=11/SEGV

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 dump_manager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dump_manager.cpp b/dump_manager.cpp
index 21fca8a..9a8d0f4 100644
--- a/dump_manager.cpp
+++ b/dump_manager.cpp
@@ -136,8 +136,8 @@ void Manager::deleteAll()
     while (iter != entries.end())
     {
         auto& entry = iter->second;
-        entry->delete_();
         ++iter;
+        entry->delete_();
     }
 }
 
-- 
2.17.1

